lib/src/repositories/ton_wallet_repository/ton_wallet_repository_impl.dart

```diff
@@ -154,8 +163,30 @@ mixin TonWalletRepositoryImpl implements TonWalletRepository {
 
   @override
   Future<void> updateSubscriptions(List<TonWalletAsset> assets) async {
-    closeAllSubscriptions();
-    for (final asset in assets) {
+    final last = lastUpdatedAssets;
+    final toSubscribe = <TonWalletAsset>[];
+    final toUnsubscribe = <TonWalletAsset>[];
+
+    if (last != null) {
+      toUnsubscribe.addAll(
+        // pick all elements from old list, which is not contains in a new list
+        last.where((asset) => !assets.any((a) => a.address == asset.address)),
+      );
+      toSubscribe.addAll(
+        // pick all elements from new list, which is not contains in old list
+        assets.where((a) => !last.any((asset) => asset.address == a.address)),
```

Comment:
Why do you swap outer and inner iteration variable (```asset``` and ```a```) in line 173 and 177?

Reply:
We have 2 similar iterations `toSubscribe` and `toUnsubscribe`. And I tried to save `asset` and `a` names for matching elements from `assets` and `last` collections

Reply:
If you want, i can change it to `a` and `l`